cmd/pyroscope/command/adhoc.go

```diff
@@ -0,0 +1,56 @@
+package command
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/pyroscope-io/pyroscope/pkg/adhoc"
+	"github.com/pyroscope-io/pyroscope/pkg/cli"
+	"github.com/pyroscope-io/pyroscope/pkg/config"
+)
+
+func newAdhocCmd(cfg *config.Adhoc) *cobra.Command {
+	vpr := newViper()
+
+	cmd := &cobra.Command{
+		Use:   "adhoc [flags] [<args>]",
+		Short: "Profile a process and save the results to be used in adhoc mode",
+		Long: `adhoc command is a complete toolset to profile a process and save the profiling
+results.
+
+These results are then available to be visualized through the 'Adhoc Profiling'
+section in the UI (available through 'pyroscope server').
+
+There are multiple ways to gather the profiling data, and not all of them are
+available for all the languages.
+Which way is better depends on several factors: what the language supports,
+how the profiled process is launched, and how the profiled process provides
+the profiled data.
```

Comment:
This is good, I think we can expand further and for each command give a suggestion for when we would use it (for example, `pull` works well with the `go` integration)

Reply:
That's a good point. I don't think we should talk about integrations in the CLI help, as integrations can evolve independently of pyroscope itself, but giving a link to the documentation may be good (and we also need to write that documentation where this is all explained, of course).